### ISSUE TITLE
how to extract/reference the audio of a video

### ISSUE THREAD

**Issue by jronallo** [IIIF/api#1332]
I'm moving a discussion from another issue onto a new issue. I'll add what I've written below. Other comments begin here on this other issue: https://github.com/IIIF/api/issues/1190#issuecomment-347654763

---

Is there a way to play only the audio from video bitstreams? Or rather what happens if the canvas has only a duration and you paint on video bitstreams/items? 

If you have a duration-only canvas, what happens if you paint on video items that target the whole canvas? Is this different from what happens when you have a duration-only canvas and paint on video items that explicitly target only the duration of the canvas through a media fragment?

The bitstreams/items do not have any indication of their height and width and so could just be treated as audio items. Also in HTML you can make a video the src of an audio element and it plays the audio just fine without displaying the video.

I agree that if you have a duration-only canvas and you try to target the canvas with a media fragment that includes a `xywh` then it should be an error since the canvas does not have any space for painting on the visual component of the video. What I'm presenting is not that case where you have a clear mismatch between explicit dimensions.

This is instead a question of what happens when you intent to reduce a video to just its duration--to just its audio. Use case is that as a manifest creator I want to use the audio from a video without having to extract the audio and publish the audio files myself, especially since I may not have the rights to redistribute the resource myself. Another use case is a video oral history where the video is boring and so the manifest creator only wants to play the audio.

---

The Media Fragments specification (basic) does mention and somewhat define the `track` dimension and give some examples, it just gives no guidance on how to handle this in HTTP or how to process the values. Some of that guidance is in the advanced version but that never got to recommendation status. 

"The way the user agent retrieves information regarding the available tracks of a particular resource is out of scope for this specification." https://www.w3.org/TR/media-frags/#media-fragment-display 
Is there any reason why we couldn't use the `track` name and specify which values a IIIF client ought to accept/reject? 

The basic specification does give guidance on media fragment servers and the case where you're retrieving audio from a video: https://www.w3.org/TR/media-frags/#media-fragment-servers
Since these are in the fragment hash of course you get back a video even if you're just requesting the audio fragment be played. No transformation of the underlying media is suggested.

If leaving off the w,h dimensions from a canvas isn't the way to play audio from a video, then it will probably be easy enough to game this by saying the canvas is 1pixel wide and high which ought to result in a player which looks like an audio player even if it is still trying to render the video. That seems silly. There's no reason to force the client to do the work to render the video component of the video if it isn't really intended to be used.

This is a common enough case that it was considered in the media fragment work. There's no transformation needed because the browser handles the case of playing just the audio from a video just fine. As far as the bitstream is concerned this is no different than muting the audio of a video and only rendering the video. This is just a case where you're "muting" the visual and only "rendering" the audio.

So one way forward would be to allow the `track` media fragment on a resource and only allow the values of `audio` and `video` (or both). The other way forward that has been mentioned in this issue is to set a time dimension only on the canvas while still having video items annotated on which would result in creation of just an audio element. How else can we achieve this? And are the use cases sufficient to drive us to find a solution?

/cc @tomcrane @workergnome @cjcolvar @azaroth42

**Comment by zimeon** [comment]
I support finding a solution for the use case of being able to reference the (an) audio from a video in order to be able to paint it onto a time-only canvas. I think we should park the selection from multiple possible audio-tracks in #1234 (this also applies in cases where we aren't taking audio only)

**Comment by tomcrane** [comment]
General agreement on AV call is to find a way to do this, to specify that you want just the audio, or just the video. It's not about named tracks in a particular resource.

A SpecifcResource with a selector:

_AudioTrackSelector_
_VideoTrackSelector_

This could be extended later for #1234

**Comment by tomcrane** [comment]
AV Call - continue discussion in Slack where we can compare JSON-LD representations

**Comment by azaroth42** [comment]
Is this impacted by #953 ?

**Comment by azaroth42** [comment]
Select seconds 10 through 30, and only play the Audio:

```
{
  "type": "Annotation",
  "body": {
    "type": "SpecificResource",
    "selector": {
      "type": "FragmentSelector",
      "value": "t=10,30"
      "refined_by": {
        "type": "AudioTrackSelector"
      }
    },
    "source": "http://uri.to/movie.mpg"
  },
  "target": {
    "id": "http://uri.to/canvas
    "type": "Canvas"
  }
}
```

**Comment by tomcrane** [comment]
IIIF AV TSG suggestions:

From video:

* x,y,t => t - Only audio (soundtrack) 
* x,y,t => x,y,t - No audio (mute) 
* x,y,t => x,y - Still image 

From audio:
 
(nothing)

```
...just the soundtrack
{
  "type": "Annotation",
  "body": {
    "type": "SpecificResource",
    "selector": {
      "type": "FragmentSelector",
      "value": "t=10,30"
      "refinedBy": {
        "type": "AudioContentSelector"
      }
    },
    "source": "http://uri.to/movie.mpg"
  },
  "target": {
    "id": "http://uri.to/canvas
    "type": "Canvas"
  }
}

... muted movie
{
  "type": "Annotation",
  "body": {
    "type": "SpecificResource",
    "selector": {
      "type": "FragmentSelector",
      "value": "t=10,30"
      "refinedBy": {
        "type": "VisualContentSelector"
      }
    },
    "source": "http://uri.to/movie.mpg"
  },
  "target": {
    "id": "http://uri.to/canvas
    "type": "Canvas"
  }
}

... screenshot of video
{
  "type": "Annotation",
  "body": {
    "type": "SpecificResource",
    "selector": {
      "type": "PointSelector",
      "value": "tp=10"
      "refinedBy": {
        "type": "VisualContentSelector"
      }
    },
    "source": "http://uri.to/movie.mpg"
  },
  "target": {
    "id": "http://uri.to/canvas
    "type": "Canvas"
  }
}

```

**Comment by azaroth42** [comment]
Editorially ... I think these join the ImageApiSelector defined at http://iiif.io/api/annex/openannotation/ and then have examples in the cookbook?

Perhaps an example in the main Prezi document for `start` referencing a time point on a canvas with duration? (per #1386)

**Comment by azaroth42** [comment]
Closed by #1500
